File: 58-length-of-last-word/length-of-last-word.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function(s) {
    s = s.trim();
    for (let i = s.length - 1; i >= 0; i--) {
        if (s[i] === ' ') {
            return s.substring(i + 1).length;
        }
    }
    return s.length;
};

File: 58-length-of-last-word/length-of-last-word.rs
impl Solution {
    pub fn length_of_last_word(s: String) -> i32 {
        let trimmed = s.trim();
        let mut length = 0;

        for c in trimmed.chars().rev() {
            if c == ' ' {
                break;
            }
            length += 1;
        }
        length
    }
}
